sdk/java/pulumi/src/main/java/com/pulumi/serialization/internal/ResourcePackages.java

```diff
@@ -159,30 +210,37 @@ static Optional<Resource> tryConstruct(String type, String version, String urn)
                         // Remove confusion of constructors with the second param of type:
                         //     Output<String> id
                         .filter(c -> !c.getParameterTypes()[1].equals(Output.class))
-                        .collect(PulumiCollectors.toSingleton(cause ->
+                        .collect(toSingleton(cause ->
                                 new IllegalArgumentException(String.format(
                                         "Resource provider error. Could not find a constructor for resource %s" +
                                                 " with the following signature:" +
                                                 " `(String name, SomeResourceArgs args, CustomResourceOptions options)`" +
                                                 ", got: `%s`",
-                                        resourceType.get(), cause))
+                                        resourceType.get(), cause
+                                ))
                         ));
 
         constructorInfo.setAccessible(true);
 
+        this.log.excessive(
+                "Deserialize/ResourcePackages: constructing using constructor=%s type=%s version=%s urn=%s",
+                constructorInfo, resourceType.get().getTypeName(), version, urn
+        );
         var resourceOptions = resolveResourceOptions(resourceType.get(), urn);
         try {
             var resource = (Resource) constructorInfo.newInstance(new Object[]{urnName, null, resourceOptions});
             return Optional.of(resource);
         } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
             throw new IllegalArgumentException(String.format(
-                    "Couldn't instantiate the '%s' class using constructor: '%s', for resource type: '%s'",
+                    "Couldn't instantiate class '%s' using constructor: '%s', for resource type: '%s'",
                     resourceType.get().getTypeName(), constructorInfo, type
             ));
+        } finally {
+            constructorInfo.setAccessible(false);
```

Comment:
I think it's this one https://docs.oracle.com/javase/10/docs/api/java/lang/reflect/AccessibleObject.html#setAccessible(boolean) and 
see https://stackoverflow.com/questions/1549804/under-what-situations-will-javas-field-setaccessibletrue-fail

If I'm reading this right, it only affects `var constructorInfo` not the class itself; and we throw away constructorInfo right after, so it is not needed. 

Also it seems that `setAccessible` may fail with SecurityException FYI. Perhaps that's ok.

Reply:
I'm just being pedantic here TBH, after setting it to `true` in line 223 above, I wanted to return it to the default position. You might be right thou about this setter working only for this `Constructor` object instead of a global state.